apps/server/src/modules/tool/external-tool/uc/external-tool.uc.spec.ts

```diff
@@ -579,4 +596,220 @@ describe('ExternalToolUc', () => {
+				externalToolService.findById.mockResolvedValue(externalTool);
+				pdfService.generatePdf.mockResolvedValueOnce(Buffer.from('mockData'));
+
+				return {
+					user,
+					toolId,
+				};
+			};
+
+			it('get user with permissions', async () => {
+				const { user, toolId } = setupDatasheetData();
+
+				await uc.getDatasheet(user.id, toolId);
+
+				expect(authorizationService.getUserWithPermissions).toHaveBeenCalledWith(user.id);
+			});
+
+			it('should check that the user has TOOL_ADMIN permission', async () => {
+				const { user, toolId } = setupDatasheetData();
+
+				await uc.getDatasheet(user.id, toolId);
+
+				expect(authorizationService.checkOneOfPermissions).toHaveBeenCalledWith(user, [
+					Permission.TOOL_ADMIN,
+					Permission.SCHOOL_TOOL_ADMIN,
+				]);
+			});
+		});
+
+		describe('when user has insufficient permission to create a datasheet for an external tool ', () => {
+			const setupDatasheetData = () => {
+				const toolId: string = new ObjectId().toHexString();
+
+				const user: User = userFactory.buildWithId();
+
+				authorizationService.getUserWithPermissions.mockRejectedValue(new UnauthorizedException());
+
+				return {
+					user,
+					toolId,
+				};
+			};
+
+			it('should throw UnauthorizedException ', async () => {
+				const { toolId, user } = setupDatasheetData();
+
+				const result = uc.getDatasheet(user.id, toolId);
+
+				await expect(result).rejects.toThrow(UnauthorizedException);
+			});
+		});
+
+		describe('when externalToolId is a basic tool', () => {
+			const setupDatasheetdata = () => {
+				const toolId: string = new ObjectId().toHexString();
+				const user: User = userFactory.buildWithId();
+
+				const externalTool: ExternalTool = externalToolFactory.withCustomParameters(1).build();
+				const params = externalTool.parameters as CustomParameter[];
+				const datasheetData: ExternalToolDatasheetTemplateData = externalToolDatasheetTemplateDataFactory
+					.withParameters(1, { name: params[0].name })
+					.build({
+						toolName: externalTool.name,
+						instance: 'dBildungscloud',
+						creatorName: `${user.firstName} ${user.lastName}`,
+					});
+
+				authorizationService.getUserWithPermissions.mockResolvedValue(user);
+				externalToolService.findById.mockResolvedValue(externalTool);
+				pdfService.generatePdf.mockResolvedValueOnce(Buffer.from('mockData'));
+
+				return {
+					user,
+					toolId,
+					datasheetData,
+				};
+			};
+
+			it('should get external tool', async () => {
+				const { toolId, user } = setupDatasheetdata();
+
+				await uc.getDatasheet(user.id, toolId);
+
+				expect(externalToolService.findById).toHaveBeenCalledWith(toolId);
+			});
+
+			it('should create pdf buffer', async () => {
+				const { toolId, user, datasheetData } = setupDatasheetdata();
+
+				await uc.getDatasheet(user.id, toolId);
+
+				expect(pdfService.generatePdf).toHaveBeenCalledWith(datasheetData);
+			});
+		});
+
+		describe('when tool is an oauth2 tool', () => {
```

Comment:
I don't think you need to test every tool type since you have tests for the mapper